libmscore/scale.cpp

```diff
@@ -0,0 +1,265 @@
+                  computeNotes35();
+                  break;
+            default:
+                  MScore::lastError = QObject::tr("Unsupported Scala file format.");
+                  return false;
+            }
+      return true;
+      }
+
+//---------------------------------------------------------
+//   computeNotes8
+//---------------------------------------------------------
+void Scale::computeNotes7() {
+      static vector<Tpc> notesToFill = {TPC_D, TPC_E, TPC_F, TPC_G, TPC_A, TPC_B, TPC_C};
+
+      for (size_t i = 0; i < originalNotes.size(); ++i) {
+            computedNotes[notesToFill[i]] = originalNotes[i];
+            }
+
+      fillMissing7();
+      }
+
+//---------------------------------------------------------
+//   computeNotes12
+//---------------------------------------------------------
+void Scale::computeNotes12() {
+      static vector<Tpc> notesToFill = {
+            TPC_C_S, TPC_D,
+            TPC_D_S, TPC_E,
+                     TPC_F,
+            TPC_F_S, TPC_G,
+            TPC_G_S, TPC_A,
+            TPC_A_S, TPC_B,
+                     TPC_C};
+
+      for (size_t i = 0; i < originalNotes.size(); ++i) {
+            computedNotes[notesToFill[i]] = originalNotes[i];
+            }
+
+      fillMissing12();
+      }
+
+//---------------------------------------------------------
+//   computeNotes17
+//---------------------------------------------------------
+void Scale::computeNotes17() {
+      static vector<Tpc> notesToFill = {
+            TPC_C_S, TPC_D_B, TPC_D,
+            TPC_D_S, TPC_E_B, TPC_E,
+                              TPC_F,
+            TPC_F_S, TPC_G_B, TPC_G,
+            TPC_G_S, TPC_A_B, TPC_A,
+            TPC_A_S, TPC_B_B, TPC_B,
+                              TPC_C};
+
+      for (size_t i = 0; i < originalNotes.size(); ++i) {
+            computedNotes[notesToFill[i]] = originalNotes[i];
+            }
+
+      fillMissing17();
+      }
+
+//---------------------------------------------------------
+//   computeNotes21
+//---------------------------------------------------------
+void Scale::computeNotes21() {
+      static vector<Tpc> notesToFill = {
+            TPC_C_S, TPC_D_B, TPC_D,
+            TPC_D_S, TPC_E_B, TPC_E,
+            TPC_E_S, TPC_F_B, TPC_F,
+            TPC_F_S, TPC_G_B, TPC_G,
+            TPC_G_S, TPC_A_B, TPC_A,
+            TPC_A_S, TPC_B_B, TPC_B,
+            TPC_B_S, TPC_C_B, TPC_C};
+
+      for (size_t i = 0; i < originalNotes.size(); ++i) {
+            computedNotes[notesToFill[i]] = originalNotes[i];
+            }
+
+      fillMissing21();
+      }
+
+//---------------------------------------------------------
+//   computeNotes35
+//---------------------------------------------------------
+void Scale::computeNotes35() {
+      static vector<Tpc> notesToFill = {
+            TPC_D_BB, TPC_C_S, TPC_D_B, TPC_C_SS, TPC_D,
+            TPC_E_BB, TPC_D_S, TPC_E_B, TPC_D_SS, TPC_E,
+            TPC_F_BB, TPC_E_S, TPC_F_B, TPC_E_SS, TPC_F,
```

Comment:
There are scala files that have 35 notes per octave (there actually are some with a lot more notes).